algorithms_task_2/student/include/impl.h

```diff
@@ -9,4 +10,10 @@
  *
  * @return true if source starts from a prefix
  */
-bool startsWith(const std::string& source, const std::string& prefix);
+bool startsWith(const std::string& source, const std::string& prefix){
+    return source.find(prefix, 0) == 0;
+    // Can also check source.begin() == std::search(source.begin(), source.end(), prefix.begin(), prefix.end());
+    // Another possibility is std::mismatch
+    // Could also check if std::equal(source.begin(), std::next(source.begin(), prefix.size), prefix.begin(), prefix.end()) is true 
```

Comment:
so why have you chosen other alternative?

Reply:
My main reason was that it would be more readable than the other ways. Complexity-wise I figured they should be almost the same.

Reply:
std::equal will return on first mismatch, so it's the most effective way